### PR TITLE
Address Sanitizer fixes (main)

### PR DESCRIPTION
**Pull request by korydraughn**
Companion PR to irods/irods#7269.

### CODE REVIEW

File: src/ilsresc.cpp [hunk 1]
```diff
@@ -490,6 +497,8 @@ main( int argc, char **argv ) {
     rcComm_t *Conn = rcConnect( myEnv.rodsHost, myEnv.rodsPort, myEnv.rodsUserName,
                       myEnv.rodsZone, 0, &errMsg );
 
+    const auto disconnect = irods::at_scope_exit{[&Conn] { rcDisconnect(Conn); }};
```
Comment (alanking) [inline src/ilsresc.cpp:500]: If `Conn` is `NULL`, we do not need to call `rcDisconnect`. Why move it here?

**Review by alanking — COMMENTED** [review]
Just one thing, pretty minor

File: src/ilsresc.cpp (hunk 1 above)
Comment (alanking) [reply src/ilsresc.cpp:500]: Got it. Thanks

Comment (korydraughn) [reply src/ilsresc.cpp:500]: I'm going to leave this as is since it's what was tested.

We can always adjust later.

**Review by alanking — COMMENTED** [review]
Please add issue numbers and squash (or don't) to taste

**Comment by korydraughn** [comment]
Attached issue number 6634 to commits.

Are we ready for pounds?

**Review by alanking — APPROVED** [review]
\# it

**Comment by korydraughn** [comment]
#'d

**Pull request merged**